pkg/contractreader/observed.go

```diff
@@ -0,0 +1,159 @@
+package contractreader
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+)
+
+var (
+	buckets = []float64{
+		float64(10 * time.Millisecond),
+		float64(20 * time.Millisecond),
+		float64(50 * time.Millisecond),
+		float64(70 * time.Millisecond),
+		float64(100 * time.Millisecond),
+		float64(200 * time.Millisecond),
+		float64(500 * time.Millisecond),
+		float64(700 * time.Millisecond),
+		float64(time.Second),
+		float64(2 * time.Second),
+		float64(5 * time.Second),
+	}
+
+	crDirectRequestsDurations = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "contract_reader_direct_request_duration",
+			Help:    "The amount of time elapsed during the ChainReader's GetLatestValue execution",
+			Buckets: buckets,
+		},
+		[]string{"chainID", "function", "contract"},
```

Comment:
Assuming 40 chains, 4 functions, and 3 contracts, this gives us 6240 new metrics. Do we really need this number of buckets?

Reply:
Yep, it's gonna be per "RPC" function called. I think this metrics is gonna be super helpful